### ISSUE TITLE
Allow turning `Cargo.toml` warnings into errors

### ISSUE THREAD

**Issue by deepak** [rust-lang/cargo#2568]
subcommand to lint format of Cargo file
and an option for all commands to fail early if format is invalid

I have not found an actual usecase to justify this
but have been learning rust and maybe this will be useful for noobs to double-check 

the smallest valid `Cargo.toml` file is 

```
[package]

name = "hello_world"
version = "0.0.1"
authors = [ "name <email@example.com>" ]
```

now let us say we leave out the name.
`cargo build` will throw an error

```
failed to parse manifest at `/src/hello-rust/Cargo.toml`

Caused by:
  expected a value of type `string` for the key `package.name`
```

will throw a warning for unused keys as well

```
[package]

name = "hello_world"
version = "0.0.1"
users = [ "name <email@example.com>" ]
```

warning "unused manifest key: package.users"

but exit status is zero and everything works

also no error is thrown for an unused section. example:

```
[package]

name = "hello_world"
version = "0.0.1"
authors = [ "Deepak Kannan <kannan.deepak@gmail.com>" ]

[dependency]
```

should have been `dependencies` rather than `dependency` above

PS: I have been using `cargo build` for all these examples. 
but am not sure, if the config is checked for unused params by each sub-command

**Comment by alexcrichton** [comment]
Sounds plausible to me! Right now we definitely don't indicate via the exit code anything about warnings, but we do so for errors! I'd be fine adding a dedicated command the specifically checked the syntax with perhaps more helpful error messages and more info communicated in the error code.

**Comment by epage** [comment]
#12115 will unblock cargo having lint levels which would allow failing on different warnings

**Comment by stinodego** [comment]
I would really love this option. As a maintainer of a big open source repository, PRs often get merged that introduce Cargo warnings like the one stated in this post. It's very annoying to have to go back and fix the issues each time. I'd like to be able to configure my CI to fail on Cargo warnings.

**Comment by epage** [comment]
Lost track of this issue.  We ended up creating #12235 which has more discussion, so I'm closing in favor of that other issue.
